### ISSUE TITLE
No database transactions implemented for copying the assignments

### ISSUE THREAD

**Issue by namanshrimali** [expertiza/expertiza#2481]
Logic for copying assignments involves copying of many aspects of an assignment and there are no transactions implemented in this logic. When there are multiple database operations carried out in a single request, they have to be carried out with transactions so if one of the operations fails, the previous changes on database should be rolled back.

Currently, if one of the operations in copying fails, the assignment still gets copied, but many aspects of it has to be updated manually, which causes issues in UI consistencies.  Assignment should only get copied if all the operations within the assignment succeeds.

**Comment by efg** [comment]
I agree that the multiple db operations should be implemented as a single transaction, which would either succeed or fail as a unit.
